File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/DrawableOptions.java
package org.espenhahn.allocate.likepaxos.registry.debug.mvc;

import java.awt.Graphics2D;
import java.awt.Stroke;

public class DrawableOptions {
	
	public Stroke stroke;
	
	public DrawableOptions(Graphics2D g2) {
		this(g2.getStroke());
	}
	
	public DrawableOptions(Stroke stroke) {
		this.stroke = stroke;
	}

	public void apply(Graphics2D g2) {
		if (stroke != null) g2.setStroke(stroke);
	}
	
}

File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/objects/DrawableOptions.kt
package org.espenhahn.allocate.likepaxos.registry.debug.mvc.objects

import java.awt.Graphics2D
import java.awt.Stroke

class DrawableOptions(var stroke: Stroke?) {

    constructor(g2: Graphics2D) : this(g2.stroke)

    fun apply(g2: Graphics2D) {
        if (this.stroke != null) g2.stroke = this.stroke
    }

}
